analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DebuggerDisplayUsesExistingMembers.vb

```diff
@@ -0,0 +1,218 @@
+﻿Imports System
+Imports System.Diagnostics
+
+Public Class TestOnPropertiesAndFields
+    Const ConstantWithoutInvalidMembers As String = "1"
+    Const ConstantWithInvalidMember As String = "{NonExisting}"
+    Const ConstantFragment1 As String = "{Non"
+    Const ConstantFragment2 As String = "Existing}"
+
+    Property SomeProperty As Integer
+    Public SomeField As Integer
+
+    <DebuggerDisplayAttribute("1")> Property WithSuffix As Integer
+    <System.Diagnostics.DebuggerDisplay("1")> Property WithNamespace As Integer
+
+    <DebuggerDisplay(Nothing)> Property WithEmptyArgList As Integer
+    <DebuggerDisplay("")> Property WithEmptyFormat As Integer
+    <DebuggerDisplay(ConstantWithoutInvalidMembers)> Property WithFormatAsConstant1 As Integer
+    <DebuggerDisplay(NameOf(ConstantWithoutInvalidMembers))> Property WithFormatAsNameOf As Integer
+
+    <DebuggerDisplay("{SomeProperty}")> Property WithExistingProperty As Integer
+    <DebuggerDisplay("{SomeField}")> Property WithExistingField As Integer
+    <DebuggerDisplay("{SomeField}")> Property WithExistingFieldVerbatim As Integer
+    <DebuggerDisplay("{1 + 1}")> Property WithNoMemberReferenced1 As Integer
+    <DebuggerDisplay("{""1"" + ""1""}")> Property WithNoMemberReferenced2 As Integer
+
+    <DebuggerDisplay("{NonExisting}")> Property WithNonExistingMember1 As Integer                          ' Noncompliant {{'NonExisting' doesn't exist in this context.}}
+    '                ^^^^^^^^^^^^^^^
+    <DebuggerDisplay("1 + {NonExisting}")> Property WithNonExistingMember2 As Integer                      ' Noncompliant {{'NonExisting' doesn't exist in this context.}}
+    '                ^^^^^^^^^^^^^^^^^^^
+    <DebuggerDisplay("{NonExisting1} bla bla {NonExisting2}")> Property WithMultipleNonExisting As Integer ' Noncompliant {{'NonExisting1' doesn't exist in this context.}}
+    '                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    <DebuggerDisplay("{NonExisting}")> Property WithNonExistingMemberVerbatim As Integer                   ' Noncompliant {{'NonExisting' doesn't exist in this context.}}
+    '                ^^^^^^^^^^^^^^^
+
+    <DebuggerDisplay(ConstantWithInvalidMember)> Property WithFormatAsConstant2 As Integer                            ' FN: constants are not checked
+    <DebuggerDisplay("{Non" & "Existing}")> Property WithFormatAsConcatenationOfLiterals As Integer                   ' FN: only simple literal supported
+    <DebuggerDisplay(ConstantFragment1 & ConstantFragment2)> Property WithFormatAsConcatenationOfConstants As Integer ' FN: only simple literal supported
+
+    <DebuggerDisplay("{this.NonExistingProperty}")> Property PropertyWithExplicitThis As Integer                      ' FN: "this." not supported
```

Comment:
The VB.NET equivalent of the C# `this `keyword is the [Me keyword](https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/program-structure/me-my-mybase-and-myclass).

Reply:
I have added one with `Me.`, and kept the `this.`, since it would be a valid expression when an instance of the class/member being decorated with `DebuggerDisplayAttribute` would be inspected by a debugger running a project in C#, even though the decorated class is written in VB.NET.

c.p.
https://learn.microsoft.com/en-us/visualstudio/debugger/using-the-debuggerdisplay-attribute?view=vs-2022

> Expressions are evaluated by the expression evaluator of the language of the current stack frame and not by the evaluator of the language in which the expression was written. This can cause unpredictable results when the languages are different.